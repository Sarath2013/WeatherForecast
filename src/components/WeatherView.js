import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CardComponent from './Card';
import BarChart from './BarChart';
import LoadingScreen from './LoadingScreen';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormControl from '@material-ui/core/FormControl';
import IconButton from '@material-ui/core/IconButton';
import ArrowBackTwoTone from '@material-ui/icons/ArrowBackTwoTone';
import ArrowForwardTwoTone from '@material-ui/icons/ArrowForwardTwoTone';
import Notify from './Notify';

class WeatherView extends Component {
   constructor(props) {
      super(props);
      this.temperatureChange = this.temperatureChange.bind(this);
      this.backwardClick = this.backwardClick.bind(this);
      this.forwardClick = this.forwardClick.bind(this);
      this.setBarChart = this.setBarChart.bind(this);
      this.materialClasses = makeStyles(theme => ({
         root: {
            flexGrow: 1,
         }
      }));
   }

   componentDidMount() {
      this.props.getWeatherData();
   }

   componentDidUpdate() {
      if (this.props.weatherData.length > 0 && !this.props.loading && this.props.curPageData.length === 0)
         this.props.getPageData(0);
   }

   forwardClick() {
      this.props.getPageData(this.props.curPageIndex + 1);
   }

   backwardClick() {
      this.props.getPageData(this.props.curPageIndex - 1);
   }

   setBarChart(cardData) {
      this.props.getActiveChartData(cardData);
   }

   temperatureChange(event) {
      this.props.getCurTempType(event.target.value);
   }

   render() {
      return (
         <div className="weather-view-container">
            {
               (this.props.curPageData.length > 0) ? (
                  <div className="weather-view">
                     <div className={this.materialClasses.root}>
                        <Grid container>
                           <Grid className="temp-type-container" item xs={12}>
                              <FormControl component="fieldset">
                                 <RadioGroup className="temp-radio-group" aria-label="Select Temperature Type" name="temperature-type" value={this.props.temperatureType} onChange={this.temperatureChange} row>
                                    <FormControlLabel
                                       value="celsius"
                                       control={<Radio color="primary" />}
                                       label="Celsius"
                                       labelPlacement="end"
                                       className="celsius"
                                    />
                                    <FormControlLabel
                                       value="fahrenheit"
                                       control={<Radio color="primary" />}
                                       label="Fahrenheit"
                                       labelPlacement="end"
                                       className="fahrenheit"
                                    />
                                 </RadioGroup>
                              </FormControl>
                           </Grid>
                           <Grid className="arrow-container left" item xs={6}>
                              {(this.props.curPageIndex > 0) ?
                                 (<IconButton color="primary" aria-label="Left Arrow" onClick={this.backwardClick}>
                                    <ArrowBackTwoTone fontSize="large" />
                                 </IconButton>) : null
                              }
                           </Grid>
                           <Grid className="arrow-container right" item xs={6}>
                              {(((this.props.curPageIndex * this.props.pageSize) + this.props.pageSize) < this.props.weatherData.length) ?
                                 (<IconButton color="primary" aria-label="Right Arrow" onClick={this.forwardClick}>
                                    <ArrowForwardTwoTone fontSize="large" />
                                 </IconButton>) : null
                              }
                           </Grid>
                           {
                              this.props.curPageData.map((obj, index) => <Grid className={(Object.keys(this.props.activeChartData).length > 0 && obj.date === this.props.activeChartData.date) ? 'card-active grid-card' : 'grid-card'} key={index} item xs={4}><CardComponent cardData={obj} onSelect={this.setBarChart} temperatureType={this.props.temperatureType} /></Grid>)
                           }
                           <Grid className="bar-chart-container" item xs={12}>
                              {
                                 (Object.keys(this.props.activeChartData).length > 0) ? <BarChart chartData={this.props.activeChartData} temperatureType={this.props.temperatureType} /> : <div className="bar-chart-notify">Please select a card to show temperature fragments in a Bar Chart.</div>
                              }

                           </Grid>
                        </Grid>
                     </div>
                  </div>) : (
                     (this.props.hasError) ?
                        <Notify classVal="alert-red" error={this.props.error} message="Something went wrong. Sorry, we can't show weather information at this time." /> :
                        (!this.props.loading && this.props.weatherData.length === 0) ? <Notify message="No weather forecasts avaialble." /> : <LoadingScreen />)
            }
         </div>);
   }
}

WeatherView.propTypes = {
   weatherData: PropTypes.arrayOf(PropTypes.shape({
      chartDataC: PropTypes.array.isRequired,
      chartDataF: PropTypes.array.isRequired,
      date: PropTypes.string.isRequired,
      displayDate: PropTypes.string.isRequired,
      humidity: PropTypes.number.isRequired,
      tempC: PropTypes.string.isRequired,
      tempF: PropTypes.string.isRequired,
      weather: PropTypes.string.isRequired,
      weatherArr: PropTypes.array.isRequired,
      fragmentsC: PropTypes.array.isRequired,
      fragmentsF: PropTypes.array.isRequired,
   })),
   curPageData: PropTypes.arrayOf(PropTypes.shape({
      chartDataC: PropTypes.array.isRequired,
      chartDataF: PropTypes.array.isRequired,
      date: PropTypes.string.isRequired,
      displayDate: PropTypes.string.isRequired,
      humidity: PropTypes.number.isRequired,
      tempC: PropTypes.string.isRequired,
      tempF: PropTypes.string.isRequired,
      weather: PropTypes.string.isRequired,
      weatherArr: PropTypes.array.isRequired,
      fragmentsC: PropTypes.array.isRequired,
      fragmentsF: PropTypes.array.isRequired,
   })),
   temperatureType: PropTypes.string.isRequired,
   curPageIndex: PropTypes.number.isRequired,
   loading: PropTypes.bool.isRequired,
   hasError: PropTypes.bool.isRequired,
   error: PropTypes.object,
   activeChartData: PropTypes.object,
   pageSize: PropTypes.number.isRequired,
   getPageData: PropTypes.func.isRequired,
   getActiveChartData: PropTypes.func.isRequired,
   getCurTempType: PropTypes.func.isRequired,
   getWeatherData: PropTypes.func.isRequired,
}

export default WeatherView;