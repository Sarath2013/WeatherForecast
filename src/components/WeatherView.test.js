import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WeatherView from './WeatherView';

let wrapper;
describe("Unit Testing - Rendering Weather view with Fahrenheit", () => {
    beforeAll(() => {
    });

    afterAll(() => {
        wrapper = null
    });

    test('renders snapshot', () => {
        wrapper = shallow(<WeatherView {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('ensure Weather view details with Fahrenheit', () => {
        expect(wrapper.instance().props.getWeatherData.mock.calls.length).toBe(1);
        expect(wrapper.find('.grid-card').length).toBe(2);
        expect(wrapper.find('.bar-chart-notify').length).toBe(1);
        expect(wrapper.find('.bar-chart-notify').props().children).toBe('Please select a card to show temperature fragments in a Bar Chart.');
        expect(wrapper.find("[name='temperature-type']").props().value).toBe('fahrenheit');
        expect(wrapper.find('.grid-card').get(0).props.children.props.cardData).toEqual(weatherData[0]);
        expect(wrapper.find('.grid-card').get(1).props.children.props.cardData).toEqual(weatherData[1]);
        expect(wrapper.instance().props.curPageData.length).toBe(2);
        expect(wrapper.instance().props.weatherData.length).toBe(2);
        expect(wrapper.instance().props.curPageData).toEqual(weatherData);
        expect(wrapper.instance().props.temperatureType).toBe('fahrenheit');
        expect(wrapper.instance().props.loading).toBe(false);
    });
});

describe("Unit Testing - Rendering Weather view with Celsius", () => {
    beforeAll(() => {
    });

    afterAll(() => {
        wrapper = null
    });

    test('renders snapshot', () => {
        wrapper = shallow(<WeatherView {...props} temperatureType="celsius" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('ensure Weather view details with Celsius', () => {
        expect(wrapper.instance().props.getWeatherData.mock.calls.length).toBe(2);
        expect(wrapper.find('.grid-card').length).toBe(2);
        expect(wrapper.find('.bar-chart-notify').length).toBe(1);
        expect(wrapper.find('.bar-chart-notify').props().children).toBe('Please select a card to show temperature fragments in a Bar Chart.');
        expect(wrapper.find("[name='temperature-type']").props().value).toBe('celsius');
        expect(wrapper.find('.grid-card').get(0).props.children.props.cardData).toEqual(weatherData[0]);
        expect(wrapper.find('.grid-card').get(1).props.children.props.cardData).toEqual(weatherData[1]);
        expect(wrapper.instance().props.curPageData.length).toBe(2);
        expect(wrapper.instance().props.weatherData.length).toBe(2);
        expect(wrapper.instance().props.curPageData).toEqual(weatherData);
        expect(wrapper.instance().props.temperatureType).toBe('celsius');
        expect(wrapper.instance().props.loading).toBe(false);
    });
});

describe("Unit Testing - Rendering Weather view with Barchart", () => {
    beforeAll(() => {
    });

    afterAll(() => {
        wrapper = null
    });

    test('renders snapshot', () => {
        wrapper = shallow(<WeatherView {...props} activeChartData={weatherData[1]} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('ensure Weather view details with Barchart', () => {
        expect(wrapper.instance().props.getWeatherData.mock.calls.length).toBe(3);
        expect(wrapper.find('.grid-card').length).toBe(2);
        expect(wrapper.find('.bar-chart-container').props().children.props.chartData).toEqual(weatherData[1]);
        expect(wrapper.instance().props.activeChartData).toEqual(weatherData[1]);
        expect(wrapper.find('.bar-chart-notify').length).toBe(0);
        expect(wrapper.find("[name='temperature-type']").props().value).toBe('fahrenheit');
        expect(wrapper.find('.grid-card').get(0).props.children.props.cardData).toEqual(weatherData[0]);
        expect(wrapper.find('.grid-card').get(1).props.children.props.cardData).toEqual(weatherData[1]);
        expect(wrapper.instance().props.curPageData.length).toBe(2);
        expect(wrapper.instance().props.weatherData.length).toBe(2);
        expect(wrapper.instance().props.curPageData).toEqual(weatherData);
        expect(wrapper.instance().props.temperatureType).toBe('fahrenheit');
        expect(wrapper.instance().props.loading).toBe(false);
    });
});

const weatherData = [{
    chartDataC: [["C", "Temp"],
    ["4C", 4],
    ["3C", 3],
    ["1C", 1]],
    chartDataF: [["F", "Temp"],
    ["39F", 39],
    ["37F", 37],
    ["33F", 33]],
    date: "2019-11-12",
    displayDate: "12 Nov 19",
    humidity: 87,
    tempC: "3C",
    tempF: "36F",
    weather: "overcast clouds",
    weatherArr: [{
        count: 2,
        weather: "overcast clouds"
    }, {
        count: 1,
        weather: "light snow"
    }],
    fragmentsC: [3.6100000000000136, 2.25, 0.3900000000000432],
    fragmentsF: [38.498000000000026, 36.05, 32.702000000000076]
},
{
    chartDataC: [["C", "Temp"],
    ["2C", 2],
    ["5C", 5],
    ["3C", 3]],
    chartDataF: [["F", "Temp"],
    ["38F", 38],
    ["32F", 32],
    ["34F", 34]],
    date: "2019-11-13",
    displayDate: "13 Nov 19",
    humidity: 89,
    tempC: "4C",
    tempF: "38F",
    weather: "light snow",
    weatherArr: [{
        count: 2,
        weather: "light snow"
    }, {
        count: 1,
        weather: "overcast clouds"
    }],
    fragmentsC: [1.6100000000000136, 4.25, 2.3900000000000432],
    fragmentsF: [39.498000000000026, 31.05, 33.702000000000076]
}];

let props = {
    weatherData: weatherData,
    curPageData: weatherData,
    temperatureType: "fahrenheit",
    curPageIndex: 0,
    loading: false,
    hasError: false,
    error: {},
    activeChartData: {},
    pageSize: 3,
    getPageData: jest.fn(),
    getActiveChartData: jest.fn(),
    getCurTempType: jest.fn(),
    getWeatherData: jest.fn(),
}
