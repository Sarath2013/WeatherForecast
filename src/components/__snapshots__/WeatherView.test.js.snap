// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing - Rendering Weather view with Barchart renders snapshot 1`] = `
<div
  className="weather-view-container"
>
  <div
    className="weather-view"
  >
    <div>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          className="temp-type-container"
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(FormControl))
            component="fieldset"
          >
            <ForwardRef(RadioGroup)
              aria-label="Select Temperature Type"
              className="temp-radio-group"
              name="temperature-type"
              onChange={[Function]}
              row={true}
              value="fahrenheit"
            >
              <WithStyles(ForwardRef(FormControlLabel))
                className="celsius"
                control={
                  <ForwardRef(WithStyles)
                    color="primary"
                  />
                }
                label="Celsius"
                labelPlacement="end"
                value="celsius"
              />
              <WithStyles(ForwardRef(FormControlLabel))
                className="fahrenheit"
                control={
                  <ForwardRef(WithStyles)
                    color="primary"
                  />
                }
                label="Fahrenheit"
                labelPlacement="end"
                value="fahrenheit"
              />
            </ForwardRef(RadioGroup)>
          </WithStyles(ForwardRef(FormControl))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="arrow-container left"
          item={true}
          xs={6}
        />
        <WithStyles(ForwardRef(Grid))
          className="arrow-container right"
          item={true}
          xs={6}
        />
        <WithStyles(ForwardRef(Grid))
          className="grid-card"
          item={true}
          key="0"
          xs={4}
        >
          <Memo(CardComponent)
            cardData={
              Object {
                "chartDataC": Array [
                  Array [
                    "C",
                    "Temp",
                  ],
                  Array [
                    "4C",
                    4,
                  ],
                  Array [
                    "3C",
                    3,
                  ],
                  Array [
                    "1C",
                    1,
                  ],
                ],
                "chartDataF": Array [
                  Array [
                    "F",
                    "Temp",
                  ],
                  Array [
                    "39F",
                    39,
                  ],
                  Array [
                    "37F",
                    37,
                  ],
                  Array [
                    "33F",
                    33,
                  ],
                ],
                "date": "2019-11-12",
                "displayDate": "12 Nov 19",
                "fragmentsC": Array [
                  3.6100000000000136,
                  2.25,
                  0.3900000000000432,
                ],
                "fragmentsF": Array [
                  38.498000000000026,
                  36.05,
                  32.702000000000076,
                ],
                "humidity": 87,
                "tempC": "3C",
                "tempF": "36F",
                "weather": "overcast clouds",
                "weatherArr": Array [
                  Object {
                    "count": 2,
                    "weather": "overcast clouds",
                  },
                  Object {
                    "count": 1,
                    "weather": "light snow",
                  },
                ],
              }
            }
            onSelect={[Function]}
            temperatureType="fahrenheit"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="card-active grid-card"
          item={true}
          key="1"
          xs={4}
        >
          <Memo(CardComponent)
            cardData={
              Object {
                "chartDataC": Array [
                  Array [
                    "C",
                    "Temp",
                  ],
                  Array [
                    "2C",
                    2,
                  ],
                  Array [
                    "5C",
                    5,
                  ],
                  Array [
                    "3C",
                    3,
                  ],
                ],
                "chartDataF": Array [
                  Array [
                    "F",
                    "Temp",
                  ],
                  Array [
                    "38F",
                    38,
                  ],
                  Array [
                    "32F",
                    32,
                  ],
                  Array [
                    "34F",
                    34,
                  ],
                ],
                "date": "2019-11-13",
                "displayDate": "13 Nov 19",
                "fragmentsC": Array [
                  1.6100000000000136,
                  4.25,
                  2.390000000000043,
                ],
                "fragmentsF": Array [
                  39.498000000000026,
                  31.05,
                  33.702000000000076,
                ],
                "humidity": 89,
                "tempC": "4C",
                "tempF": "38F",
                "weather": "light snow",
                "weatherArr": Array [
                  Object {
                    "count": 2,
                    "weather": "light snow",
                  },
                  Object {
                    "count": 1,
                    "weather": "overcast clouds",
                  },
                ],
              }
            }
            onSelect={[Function]}
            temperatureType="fahrenheit"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="bar-chart-container"
          item={true}
          xs={12}
        >
          <BarChart
            chartData={
              Object {
                "chartDataC": Array [
                  Array [
                    "C",
                    "Temp",
                  ],
                  Array [
                    "2C",
                    2,
                  ],
                  Array [
                    "5C",
                    5,
                  ],
                  Array [
                    "3C",
                    3,
                  ],
                ],
                "chartDataF": Array [
                  Array [
                    "F",
                    "Temp",
                  ],
                  Array [
                    "38F",
                    38,
                  ],
                  Array [
                    "32F",
                    32,
                  ],
                  Array [
                    "34F",
                    34,
                  ],
                ],
                "date": "2019-11-13",
                "displayDate": "13 Nov 19",
                "fragmentsC": Array [
                  1.6100000000000136,
                  4.25,
                  2.390000000000043,
                ],
                "fragmentsF": Array [
                  39.498000000000026,
                  31.05,
                  33.702000000000076,
                ],
                "humidity": 89,
                "tempC": "4C",
                "tempF": "38F",
                "weather": "light snow",
                "weatherArr": Array [
                  Object {
                    "count": 2,
                    "weather": "light snow",
                  },
                  Object {
                    "count": 1,
                    "weather": "overcast clouds",
                  },
                ],
              }
            }
            temperatureType="fahrenheit"
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </div>
  </div>
</div>
`;

exports[`Unit Testing - Rendering Weather view with Celsius renders snapshot 1`] = `
<div
  className="weather-view-container"
>
  <div
    className="weather-view"
  >
    <div>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          className="temp-type-container"
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(FormControl))
            component="fieldset"
          >
            <ForwardRef(RadioGroup)
              aria-label="Select Temperature Type"
              className="temp-radio-group"
              name="temperature-type"
              onChange={[Function]}
              row={true}
              value="celsius"
            >
              <WithStyles(ForwardRef(FormControlLabel))
                className="celsius"
                control={
                  <ForwardRef(WithStyles)
                    color="primary"
                  />
                }
                label="Celsius"
                labelPlacement="end"
                value="celsius"
              />
              <WithStyles(ForwardRef(FormControlLabel))
                className="fahrenheit"
                control={
                  <ForwardRef(WithStyles)
                    color="primary"
                  />
                }
                label="Fahrenheit"
                labelPlacement="end"
                value="fahrenheit"
              />
            </ForwardRef(RadioGroup)>
          </WithStyles(ForwardRef(FormControl))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="arrow-container left"
          item={true}
          xs={6}
        />
        <WithStyles(ForwardRef(Grid))
          className="arrow-container right"
          item={true}
          xs={6}
        />
        <WithStyles(ForwardRef(Grid))
          className="grid-card"
          item={true}
          key="0"
          xs={4}
        >
          <Memo(CardComponent)
            cardData={
              Object {
                "chartDataC": Array [
                  Array [
                    "C",
                    "Temp",
                  ],
                  Array [
                    "4C",
                    4,
                  ],
                  Array [
                    "3C",
                    3,
                  ],
                  Array [
                    "1C",
                    1,
                  ],
                ],
                "chartDataF": Array [
                  Array [
                    "F",
                    "Temp",
                  ],
                  Array [
                    "39F",
                    39,
                  ],
                  Array [
                    "37F",
                    37,
                  ],
                  Array [
                    "33F",
                    33,
                  ],
                ],
                "date": "2019-11-12",
                "displayDate": "12 Nov 19",
                "fragmentsC": Array [
                  3.6100000000000136,
                  2.25,
                  0.3900000000000432,
                ],
                "fragmentsF": Array [
                  38.498000000000026,
                  36.05,
                  32.702000000000076,
                ],
                "humidity": 87,
                "tempC": "3C",
                "tempF": "36F",
                "weather": "overcast clouds",
                "weatherArr": Array [
                  Object {
                    "count": 2,
                    "weather": "overcast clouds",
                  },
                  Object {
                    "count": 1,
                    "weather": "light snow",
                  },
                ],
              }
            }
            onSelect={[Function]}
            temperatureType="celsius"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="grid-card"
          item={true}
          key="1"
          xs={4}
        >
          <Memo(CardComponent)
            cardData={
              Object {
                "chartDataC": Array [
                  Array [
                    "C",
                    "Temp",
                  ],
                  Array [
                    "2C",
                    2,
                  ],
                  Array [
                    "5C",
                    5,
                  ],
                  Array [
                    "3C",
                    3,
                  ],
                ],
                "chartDataF": Array [
                  Array [
                    "F",
                    "Temp",
                  ],
                  Array [
                    "38F",
                    38,
                  ],
                  Array [
                    "32F",
                    32,
                  ],
                  Array [
                    "34F",
                    34,
                  ],
                ],
                "date": "2019-11-13",
                "displayDate": "13 Nov 19",
                "fragmentsC": Array [
                  1.6100000000000136,
                  4.25,
                  2.390000000000043,
                ],
                "fragmentsF": Array [
                  39.498000000000026,
                  31.05,
                  33.702000000000076,
                ],
                "humidity": 89,
                "tempC": "4C",
                "tempF": "38F",
                "weather": "light snow",
                "weatherArr": Array [
                  Object {
                    "count": 2,
                    "weather": "light snow",
                  },
                  Object {
                    "count": 1,
                    "weather": "overcast clouds",
                  },
                ],
              }
            }
            onSelect={[Function]}
            temperatureType="celsius"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="bar-chart-container"
          item={true}
          xs={12}
        >
          <div
            className="bar-chart-notify"
          >
            Please select a card to show temperature fragments in a Bar Chart.
          </div>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </div>
  </div>
</div>
`;

exports[`Unit Testing - Rendering Weather view with Fahrenheit renders snapshot 1`] = `
<div
  className="weather-view-container"
>
  <div
    className="weather-view"
  >
    <div>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          className="temp-type-container"
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(FormControl))
            component="fieldset"
          >
            <ForwardRef(RadioGroup)
              aria-label="Select Temperature Type"
              className="temp-radio-group"
              name="temperature-type"
              onChange={[Function]}
              row={true}
              value="fahrenheit"
            >
              <WithStyles(ForwardRef(FormControlLabel))
                className="celsius"
                control={
                  <ForwardRef(WithStyles)
                    color="primary"
                  />
                }
                label="Celsius"
                labelPlacement="end"
                value="celsius"
              />
              <WithStyles(ForwardRef(FormControlLabel))
                className="fahrenheit"
                control={
                  <ForwardRef(WithStyles)
                    color="primary"
                  />
                }
                label="Fahrenheit"
                labelPlacement="end"
                value="fahrenheit"
              />
            </ForwardRef(RadioGroup)>
          </WithStyles(ForwardRef(FormControl))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="arrow-container left"
          item={true}
          xs={6}
        />
        <WithStyles(ForwardRef(Grid))
          className="arrow-container right"
          item={true}
          xs={6}
        />
        <WithStyles(ForwardRef(Grid))
          className="grid-card"
          item={true}
          key="0"
          xs={4}
        >
          <Memo(CardComponent)
            cardData={
              Object {
                "chartDataC": Array [
                  Array [
                    "C",
                    "Temp",
                  ],
                  Array [
                    "4C",
                    4,
                  ],
                  Array [
                    "3C",
                    3,
                  ],
                  Array [
                    "1C",
                    1,
                  ],
                ],
                "chartDataF": Array [
                  Array [
                    "F",
                    "Temp",
                  ],
                  Array [
                    "39F",
                    39,
                  ],
                  Array [
                    "37F",
                    37,
                  ],
                  Array [
                    "33F",
                    33,
                  ],
                ],
                "date": "2019-11-12",
                "displayDate": "12 Nov 19",
                "fragmentsC": Array [
                  3.6100000000000136,
                  2.25,
                  0.3900000000000432,
                ],
                "fragmentsF": Array [
                  38.498000000000026,
                  36.05,
                  32.702000000000076,
                ],
                "humidity": 87,
                "tempC": "3C",
                "tempF": "36F",
                "weather": "overcast clouds",
                "weatherArr": Array [
                  Object {
                    "count": 2,
                    "weather": "overcast clouds",
                  },
                  Object {
                    "count": 1,
                    "weather": "light snow",
                  },
                ],
              }
            }
            onSelect={[Function]}
            temperatureType="fahrenheit"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="grid-card"
          item={true}
          key="1"
          xs={4}
        >
          <Memo(CardComponent)
            cardData={
              Object {
                "chartDataC": Array [
                  Array [
                    "C",
                    "Temp",
                  ],
                  Array [
                    "2C",
                    2,
                  ],
                  Array [
                    "5C",
                    5,
                  ],
                  Array [
                    "3C",
                    3,
                  ],
                ],
                "chartDataF": Array [
                  Array [
                    "F",
                    "Temp",
                  ],
                  Array [
                    "38F",
                    38,
                  ],
                  Array [
                    "32F",
                    32,
                  ],
                  Array [
                    "34F",
                    34,
                  ],
                ],
                "date": "2019-11-13",
                "displayDate": "13 Nov 19",
                "fragmentsC": Array [
                  1.6100000000000136,
                  4.25,
                  2.390000000000043,
                ],
                "fragmentsF": Array [
                  39.498000000000026,
                  31.05,
                  33.702000000000076,
                ],
                "humidity": 89,
                "tempC": "4C",
                "tempF": "38F",
                "weather": "light snow",
                "weatherArr": Array [
                  Object {
                    "count": 2,
                    "weather": "light snow",
                  },
                  Object {
                    "count": 1,
                    "weather": "overcast clouds",
                  },
                ],
              }
            }
            onSelect={[Function]}
            temperatureType="fahrenheit"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="bar-chart-container"
          item={true}
          xs={12}
        >
          <div
            className="bar-chart-notify"
          >
            Please select a card to show temperature fragments in a Bar Chart.
          </div>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </div>
  </div>
</div>
`;
